dev/src/webpack/getDefaultWebpackConfig.ts

```diff
@@ -0,0 +1,288 @@
+import * as TerserPlugin from 'terser-webpack-plugin';
+import * as webpack from 'webpack';
+import { Verbosity } from '../..';
+import { DefaultWebpackOptions } from '../../index';
+import { excludeNodeModulesAndDependencies, PackageLock } from './excludeNodeModulesAndDependencies';
+
+// Using webpack helps reduce the install and startup time of large extensions by reducing the large number of files into a much smaller set
+// Full webpack documentation: [https://webpack.js.org/configuration/]().
+
+// tslint:disable:no-any // A lot of the plug-ins use functions with any arguments
+
+type MessageVerbosity = Exclude<Verbosity, 'silent'>;
+
+const verbosityMap: Map<Verbosity, number> = new Map<Verbosity, number>();
+verbosityMap.set('silent', 0);
+verbosityMap.set('normal', 1);
+verbosityMap.set('debug', 2);
+
+const defaultExternalNodeModules: string[] = [
+    'vscode-languageclient'
+];
+
+// tslint:disable-next-line:max-func-body-length
+export function getDefaultWebpackConfig(options: DefaultWebpackOptions): webpack.Configuration {
+    // tslint:disable-next-line: strict-boolean-expressions
+    const loggingVerbosity: Verbosity = options.verbosity || 'normal';
+
+    // Only use defaultExternalNodeModules entries that are actually in package-lock.json
+    const packageLockJson: PackageLock = fse.readJsonSync(path.resolve(options.projectRoot, 'package-lock.json'));
+    // tslint:disable-next-line: strict-boolean-expressions
+    const existingDefaultExtNodeModules: string[] = defaultExternalNodeModules.filter((moduleName: string) => !!packageLockJson[moduleName]);
+
+    // tslint:disable-next-line: strict-boolean-expressions
+    const externalNodeModules: string[] = (options.externalNodeModules || []).concat(existingDefaultExtNodeModules);
+    log('debug', 'externalNodeModules:', externalNodeModules);
+
+    function log(messageVerbosity: MessageVerbosity, ...args: unknown[]): void {
+        logCore(loggingVerbosity, messageVerbosity, ...args);
+    }
+
+    const config: webpack.Configuration = {
+        context: options.projectRoot,
+
+        // vscode extensions run in a Node.js context, see https://webpack.js.org/configuration/node/
+        target: 'node',
+        node: {
+            // For __dirname and __filename, let Node.js use its default behavior (i.e., gives the path to the packed extension.js file, not the original source file)
+            __filename: false,
+            __dirname: false
+        },
+
+        entry: {
+            // Note: Each entry is a completely separate Node.js application that cannot interact with any
+            // of the others, and that individually includes all dependencies necessary (i.e. common
+            // dependencies will have a copy in each entry file, no sharing).
+
+            // The entrypoint bundle for this extension, see https://webpack.js.org/configuration/entry-context/
+            extension: './extension.ts',
```

Comment:
Would prefer to finalize this conversation about names before other repos start using this, but otherwise PR looks good
https://github.com/Microsoft/vscode-azureappservice/pull/828#discussion_r249152159

Reply:
Settled.